runtime/str.go

```diff
@@ -157,6 +157,29 @@ func strAdd(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return NewStr(stringV + stringW).ToObject(), nil
 }
 
+func strCapitalize(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "capitalize", args, StrType); raised != nil {
+		return nil, raised
+	}
+	s := toStrUnsafe(args[0]).Value()
+	buf := bytes.Buffer{}
+	runes := []rune(s)
+	numRunes := len(runes)
+	writeRune := func(r rune, transform func(r rune) rune) {
+		if r <= unicode.MaxASCII {
+			r = transform(r)
+		}
+		buf.WriteRune(r)
+	}
+	if numRunes > 0 {
+		writeRune(runes[0], func(r rune) rune { return unicode.ToUpper(r) })
+	}
+	for i := 1; i < numRunes; i++ {
```

Comment:
OK, I think I understand. Because you're only modifying chars < 128, `в` (U+1074) isn't affected. If a multibyte character contained bytes < 128 then CPython would do something differe, but I *think* that's never true in UTF-8. So you're right, in fact, this code should work but kind of by coincidence. It also seems a bit roundabout considering all we're interested in converting are the ASCII chars.

IMO the simplest and most efficient way to implement this are two tables for mapping lowercase ASCII chars to uppercase and vice versa. But you could also do it by checking `s[i] >= 'A' && s[i] <= 'Z'` and then setting `s[i] = s[i] - 'A' + 'a'` to go lowercase (and similar for uppercase).